src/Utils/TrackScheduler.cpp

```diff
@@ -0,0 +1,47 @@
+/*****************************************************************************/
+/**
+ * @file    TrackScheduler.cpp
+ * @author  Michael Wurm <wurm.michael95@gmail.com>
+ * @brief   Class TrackScheduler implementation
+ */
+/*****************************************************************************/
+
+#include "TrackScheduler.h"
+
+using namespace std;
+
+TrackScheduler::TrackScheduler(DataStore* const datastore,
+                               MusicBackend* const musicbackend) {
+  if (datastore == nullptr)
+    LOG(ERROR) << "TrackScheduler Ctor: datastore is nullptr!";
+  if (musicbackend == nullptr)
+    LOG(ERROR) << "TrackScheduler Ctor: datastore is nullptr!";
+
+  mDataStore = datastore;
+  mMusicBackend = musicbackend;
+}
+
+TrackScheduler::~TrackScheduler() {
+  mThread.join();
+
+  /* Memory needs to be freed by the creator of this object.
+   * Just de-initializing the pointers here for safety reasons.
+   * TODO: Discuss if that makes sense :)
```

Comment:
if the class doesn't create the objects it isn't responsible for destroying them. since `TrackScheduler` will never outlive the creating class (`JukeBox`) this is perfectly fine